### PR TITLE
chore: update to Core Crypto 0.5.0 FS-963

### PR DESCRIPTION
**Pull request by johnxnguyen**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-963" title="FS-963" target="_blank">TaskFS-963</a>  [iOS] Update to Core Crypto 0.5.1
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- Update Core Crypto to 0.5.0.
- Refactor how we mock Core Crypto: 
  - Instead of providing a mock result or a mock error, provide a mock function
  - Remove the `Calls` struct that counts the invocations of methods... it's hard to maintain and not necessary, since we can make assertions in the mock functions.
- Update tests to account for the new mock approach.

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: MLS/MLSController.swift [hunk 1]
```diff
@@ -57,13 +57,14 @@ class DummyCoreCryptoCallbacks: CoreCryptoCallbacks {
...
     init() {}
 
-    func authorize(conversationId: [UInt8], clientId: String) -> Bool {
+    func authorize(conversationId: [UInt8], clientId: [UInt8]) -> Bool {
         return true
     }
 
-    func isUserInGroup(identity: [UInt8], otherClients: [[UInt8]]) -> Bool {
+    func clientIdBelongsToOneOf(clientId: [UInt8], otherClients: [[UInt8]]) -> Bool {
         return true
```
Comment (typfel) [inline MLS/MLSController.swift:65]: This we actually need to implement it's necessary for verification external join proposals. We need check that the qualified user of`clientId` exists in the list of `otherClients`.

You can of course do this in a separate PR.

**Review by typfel — APPROVED** [review]

File: MLS/MLSController.swift (hunk 1 above)
Comment (johnxnguyen) [reply MLS/MLSController.swift:65]: > We need check that the qualified user ofclientId exists in the list of otherClients.

@typfel  I don't quite understand what you mean.

Comment (typfel) [reply MLS/MLSController.swift:65]: clientId is a qualified client ID string ("userid:clientid:domain" )
otherClients is a list of qualified client IDs

we should check the userId + domain exists in the otherClients list. In other words does otherClients already contain another client from this user.

**Review by David-Henner — APPROVED** [review]

File: MLS/CoreCryptoTypes.swift [hunk 2]
```diff
@@ -34,33 +34,40 @@ public protocol CoreCryptoProtocol {
...
     func wire_conversationExists(conversationId: ConversationId)  -> Bool
     func wire_processWelcomeMessage(welcomeMessage: [UInt8]) throws -> ConversationId
     func wire_addClientsToConversation(conversationId: ConversationId, clients: [Invitee]) throws -> MemberAddedMessages
     func wire_removeClientsFromConversation(conversationId: ConversationId, clients: [ClientId]) throws -> CommitBundle
+    func wire_updateKeyingMaterial(conversationId: ConversationId) throws -> CommitBundle
+    func wire_commitPendingProposals(conversationId: ConversationId) throws -> CommitBundle?
+    func wire_finalAddClientsToConversation(conversationId: ConversationId, clients: [Invitee]) throws -> TlsCommitBundle
+    func wire_finalRemoveClientsFromConversation(conversationId: ConversationId, clients: [ClientId]) throws -> TlsCommitBundle
+    func wire_finalUpdateKeyingMaterial(conversationId: ConversationId) throws -> TlsCommitBundle
+    func wire_finalCommitPendingProposals(conversationId: ConversationId) throws -> TlsCommitBundle?
```
[the remark this answers is not in the document]
Comment (typfel) [reply MLS/CoreCryptoTypes.swift:47]: It's an API which we will use later in M5, I guess the names might change later when existing methods get obsolete. CoreCrypto just wanted API already in there.

File: MLS/MLSController.swift (hunk 1 above)
Comment (johnxnguyen) [reply MLS/MLSController.swift:65]: Why does the iOS client need to do this check? It seems like something Core Crypto could do.
